std/internal/math/biguintcore.d

```diff
@@ -1402,59 +1402,76 @@ void mulInternal(BigDigit[] result, const(BigDigit)[] x, const(BigDigit)[] y)
         auto extra =  x.length % y.length;
         auto maxchunk = chunksize + extra;
         bool paddingY; // true = we're padding Y, false = we're padding X.
-        if (extra * extra * 2 < y.length*y.length)
-        {
-            // The leftover bit is small enough that it should be incorporated
-            // in the existing chunks.
-            // Make all the chunks a tiny bit bigger
-            // (We're padding y with zeros)
-            chunksize += extra / numchunks;
-            extra = x.length - chunksize*numchunks;
-            // there will probably be a few left over.
-            // Every chunk will either have size chunksize, or chunksize+1.
-            maxchunk = chunksize + 1;
-            paddingY = true;
-            assert(chunksize + extra + chunksize *(numchunks-1) == x.length );
+        bool isExtraSmall = extra * extra * 2 < y.length * y.length;
+        if (numchunks == 1 && isExtraSmall)
+        {
+            // We divide (x_first_half * y) and (x_last_half * y)
+            // between 1.414:1 and 1.707:1 (1.707 = 1+1/sqrt(2)).
+            // (1.414 ~ 1.707)/2:1 is balanced.
+            BigDigit [] scratchbuff = new BigDigit[karatsubaRequiredBuffSize(y.length) + y.length];
+            BigDigit [] partial = scratchbuff[$ - y.length .. $];
+            mulKaratsuba(result[0 .. half + y.length], y, x[0 .. half], scratchbuff);
+            partial[] = result[half .. half + y.length];
+            mulKaratsuba(result[half .. $], y, x[half .. $], scratchbuff);
+            addAssignSimple(result[half .. half + y.length], partial);
```

Comment:
This looks structurally very similar to the variant below, any chance this could be integrated there?